### PR TITLE
Update Moveable when resizing window

### LINKED ISSUE #2154: Multi-selection box is not updated/responsive
## Bug Description

The multi-selection box is not updated when switched between small/big canvas.

## Expected Behaviour

The selection box should update after the canvas is resized.

## Steps to Reproduce

- Select 2+ elements while on small/big canvas.
- Resize to big/small canvas.
- The selection box is not updated.

## Screenshots

responsiveLasso

## Additional Context

- Plugin Version:
- Operating System:
- Browser:

---

_Do not alter or remove anything below. The following sections will be managed by moderators only._

### LINKED ISSUE #2785: Selection frame out of place when  resizing window
## Bug Description

Originally reported by @samitron7 on Slack.

## Expected Behaviour

The selection frame should move with the element.

## Steps to Reproduce

1. Create element
1. Make sure element is selected
1. Resize browser window
1. See how selection frame is not at the same place anymore

## Screenshots

ezgif-4-e97c17f11cd3

## Additional Context

- Plugin Version: `master`
- Operating System:
- Browser:

---

_Do not alter or remove anything below. The following sections will be managed by moderators only._

### PR DESCRIPTION
**Pull request by miina**
## Summary
Adds listener for window resize event to update moveable.

## Testing Instructions
I
1. Add an element and select it.
2. Resize the window. Verify that the blue selection line moves correctly together with the element while resizing.

II
1. Add multiple elements and select all.
2. Resize the window. Verify that the blue selection line moves correctly together with the elements while resizing.

---

Fixes #2785.
Fixes #2154

### CODE REVIEW

File: assets/src/edit-story/components/canvas/multiSelectionMoveable/index.js [hunk 1]
```diff
@@ -65,6 +65,16 @@ function MultiSelectionMoveable({ selectedElements }) {
     actions: { pushTransform },
   } = useTransform();
 
+  useEffect(() => {
+    const handleUpdateMoveable = () => {
+      moveable.current?.updateRect();
+    };
```
Comment (barklund) [inline assets/src/edit-story/components/canvas/multiSelectionMoveable/index.js]: Could just be:

```js
    const handleUpdateMoveable = () => moveable.current?.updateRect();
```

File: assets/src/edit-story/components/canvas/singleSelectionMoveable/index.js [hunk 2]
```diff
@@ -65,6 +65,16 @@ function SingleSelectionMoveable({
...
 
+  useEffect(() => {
+    const handleUpdateMoveable = () => {
+      moveable.current?.updateRect();
+    };
+    window.addEventListener('resize', handleUpdateMoveable);
+    return () => {
+      window.removeEventListener('resize', handleUpdateMoveable);
+    };
+  }, [moveable]);
```
Comment (barklund) [inline assets/src/edit-story/components/canvas/singleSelectionMoveable/index.js]: We could also move it to a reusable hook: `useResizeableMovable(ref)` or similar?

**Review by barklund — APPROVED** [review]
Works great, just has a few opportunities for improvement.

File: assets/src/edit-story/components/canvas/multiSelectionMoveable/index.js [hunk 3]
```diff
@@ -65,6 +65,16 @@ function MultiSelectionMoveable({ selectedElements }) {
...
+    window.addEventListener('resize', handleUpdateMoveable);
+    return () => {
+      window.removeEventListener('resize', handleUpdateMoveable);
+    };
```
Comment (barklund) [inline assets/src/edit-story/components/canvas/multiSelectionMoveable/index.js]: Oh, actually this one can be compressed too.

File: assets/src/edit-story/components/canvas/multiSelectionMoveable/index.js (hunk 1 above)
Comment (barklund) [reply assets/src/edit-story/components/canvas/multiSelectionMoveable/index.js]: And if it wraps in two lines still, the function name can be shortened significantly to just `update` or similar.

File: assets/src/edit-story/components/canvas/singleSelectionMoveable/index.js (hunk 2 above)
Comment (miina) [reply assets/src/edit-story/components/canvas/singleSelectionMoveable/index.js]: Hm, wasn't certain if it's worth it in this case but will move if you think it's better!

**Pull request merged**